server/container_create.go

```diff
@@ -388,34 +388,14 @@ func buildOCIProcessArgs(containerKubeConfig *pb.ContainerConfig, imageOCIConfig
 	kubeArgs := containerKubeConfig.Args
 
 	// merge image config and kube config
-	// same as docker does today...
-	if imageOCIConfig != nil {
-		if len(kubeCommands) == 0 {
-			if len(kubeArgs) == 0 {
-				kubeArgs = imageOCIConfig.Config.Cmd
-			}
-			if kubeCommands == nil {
-				kubeCommands = imageOCIConfig.Config.Entrypoint
-			}
+	if ociConfig != nil && len(kubeCommands) == 0 {
+		kubeCommands = ociConfig.Entrypoint
+		if len(kubeArgs) == 0 {
+			kubeArgs = ociConfig.Cmd
 		}
 	}
 
-	if len(kubeCommands) == 0 && len(kubeArgs) == 0 {
-		return nil, fmt.Errorf("no command specified")
-	}
-
-	// create entrypoint and args
-	var entrypoint string
-	var args []string
-	if len(kubeCommands) != 0 {
```

Comment:
We will fail tests and regress on behavior if we get rid of this entire block.

Reply:
> We will fail tests and regress on behavior if we get rid of this entire block.

d3d9de5c7 is just about this change, and I tried to motivate it in the commit message.  [Here][1] is a Go example showing that this change is a functional no-op for cases where some arguments are set (and the old code has a guard for no arguments being set, but that's 9574d0e1).  Can you give an example of what d3d9de5c7 breaks?

[1]: https://play.golang.org/p/ek9YxY4L5qP